src/evo/providertx.cpp

```diff
@@ -247,6 +257,87 @@ void ProRegPL::ToJson(UniValue& obj) const
     obj.pushKV("inputsHash", inputsHash.ToString());
 }
 
+// Provider Update Service Payload
+
+bool CheckProUpServTx(const CTransaction& tx, const CBlockIndex* pindexPrev, CValidationState& state)
+{
+    assert(tx.nType == CTransaction::TxType::PROUPSERV);
+
+    ProUpServPL pl;
+    if (!GetTxPayload(tx, pl)) {
+        return state.DoS(100, false, REJECT_INVALID, "bad-protx-payload");
+    }
+
+    if (pl.nVersion == 0 || pl.nVersion > ProUpServPL::CURRENT_VERSION) {
+        return state.DoS(100, false, REJECT_INVALID, "bad-protx-version");
+    }
+
+    if (!CheckService(pl.addr, state)) {
+        // pass the state returned by the function above
+        return false;
+    }
+
+    if (!CheckInputsHash(tx, pl, state)) {
+        // pass the state returned by the function above
+        return false;
+    }
+
+    if (pindexPrev) {
+        auto mnList = deterministicMNManager->GetListForBlock(pindexPrev);
+        auto mn = mnList.GetMN(pl.proTxHash);
+        if (!mn) {
+            return state.DoS(100, false, REJECT_INVALID, "bad-protx-hash");
+        }
+
+        // don't allow updating to addresses already used by other MNs
+        if (mnList.HasUniqueProperty(pl.addr) && mnList.GetUniquePropertyMN(pl.addr)->proTxHash != pl.proTxHash) {
+            return state.DoS(10, false, REJECT_DUPLICATE, "bad-protx-dup-addr");
+        }
+
+        if (pl.scriptOperatorPayout != CScript()) {
```

Comment:
Better to use `!pl.scriptOperatorPayout.empty()`

Reply:
:+1: done.